vhost/src/vhost_user/message.rs

```diff
@@ -990,6 +994,48 @@ impl VhostUserMsgValidator for VhostUserTransferDeviceState {
     }
 }
 
+// Bit mask for flags in VhostUserBackendMapMsg backend message
+bitflags! {
+    #[derive(Copy, Clone, Debug, Eq, PartialEq, Default)]
+    /// Flags for virtio-fs backend messages.
+    pub struct VhostUserMMapFlags: u64 {
+        /// Empty permission.
+        const EMPTY = 0x0;
+        /// Read permission.
+        const MAP_R = 0x1;
+        /// Write permission.
+        const MAP_W = 0x2;
+    }
+}
+
+/// Backend request to mmap a file-backed buffer into guest memory
+#[repr(C, packed)]
+#[derive(Debug, Copy, Clone, Default)]
+pub struct VhostUserMMap {
+    /// Shared memory region ID.
+    pub shmid: u8,
+    /// Struct padding.
+    pub padding: [u8; 7],
+    /// File offset.
+    pub fd_offset: u64,
+    /// Offset into the shared memory region.
+    pub shm_offset: u64,
+    /// Size of region to map.
+    pub len: u64,
+    /// Flags for the mmap operation
+    pub flags: VhostUserMMapFlags,
```

Comment:
I'd follow what we did for other struct like `struct VhostUserMsgHeader` and `struct VhostUserHeaderFlag`. So here I'd use `u64`, also because in the `VhostUserMsgValidator` implementation you're already calling `.bits()`.